client/src/blessed.ts

```diff
@@ -189,8 +190,10 @@ export const renderScreenView = (gameState: GameState) => {
         gameState.marbleClicked = node;
         setCurrentState(gameState);
 
-        await postGameState(gameState);
-        renderScreenView(gameState);
+        if (PLAYER_ID === gameState.currentPlayer.playerNumber) {
+          await postGameState(gameState);
+          renderScreenView(gameState);
```

Comment:
In this case you wait that the postGameState has been correctly resolve before rendering this.
You also receive a WS event with this game state probably at the same moment no ? I'm not sure of my advice here ^^ 

It also can be useful to render a change like optimistic rendering (aka before sending any result to the API) to provide a more UX to your players.